pkg/scheduler/bind-plugin.go

```diff
@@ -0,0 +1,138 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	"placement-request-controller/pkg/apis/v1alpha1"
+	"placement-request-controller/pkg/generated/clientset/versioned"
+)
+
+// this global variable is used to ensure, at compile time, that the BindPlugin
+// struct complies with the expected framework interface.
+var _ framework.BindPlugin = &BindPlugin{}
+
+// BindPlugin implements the framework.BindPlugin interface for binding pods to
+// nodes. Its purpose is to generate PlacementRequest for pods and wait until
+// they are done.
+type BindPlugin struct {
+	client versioned.Interface
+}
+
+// Name purpose is to return the plugin name so the scheduler framework can
+// identify it while parsing the configuration.
+func (p *BindPlugin) Name() string {
+	return PluginName
+}
+
+// Bind purpose is to bind the provided pod to the specified, by name, node. It
+// should return if the bound succeeded or not so it needs to wait for the
+// PlacementRequest to be fulfilled by the placement request controller.
+func (p *BindPlugin) Bind(
+	ctx context.Context, state *framework.CycleState, pod *corev1.Pod, node string,
+) *framework.Status {
+	pr := &v1alpha1.PlacementRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pod.Name,
+			Namespace: pod.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: corev1.SchemeGroupVersion.String(),
+					Kind:       "Pod",
+					Name:       pod.Name,
+					UID:        pod.UID,
+				},
+			},
+		},
+		Spec: v1alpha1.PlacementRequestSpec{
+			Policy:        v1alpha1.PlacementRequestPolicyLenient,
+			Priority:      0,
+			SchedulerName: SchedulerName,
```

Comment:
The scheduler name needs to be configurable so we can deploy multiple instances of the same image.